base/random/RNGs.jl

```diff
@@ -0,0 +1,424 @@
+        # this is similar for the 64 high bits of u
+        A128[i] = mask128(u, T)
+    end
+    for i in 16*n128÷sizeof(T)+1:n
+        @inbounds A[i] = rand(r, T) + oneunit(T)
+    end
+    A
+end
+
+function rand!(r::MersenneTwister, A::Union{Array{Float16},Array{Float32}},
+               ::Type{CloseOpen})
+    rand!(r, A, Close1Open2)
+    I32 = one(Float32)
+    for i in eachindex(A)
+        @inbounds A[i] = Float32(A[i])-I32 # faster than "A[i] -= one(T)" for T==Float16
+    end
+    A
+end
+
+rand!(r::MersenneTwister, A::Union{Array{Float16},Array{Float32}}) = rand!(r, A, CloseOpen)
+
+function rand!(r::MersenneTwister, A::Array{UInt128}, n::Int=length(A))
+    if n > length(A)
+        throw(BoundsError(A,n))
+    end
+    Af = unsafe_wrap(Array, convert(Ptr{Float64}, pointer(A)), 2n)
+    i = n
+    while true
+        rand!(r, Af, 2i, Close1Open2)
+        n < 5 && break
+        i = 0
+        @inbounds while n-i >= 5
+            u = A[i+=1]
+            A[n]    ⊻= u << 48
+            A[n-=1] ⊻= u << 36
+            A[n-=1] ⊻= u << 24
+            A[n-=1] ⊻= u << 12
+            n-=1
+        end
+    end
+    if n > 0
+        u = rand_ui2x52_raw(r)
+        for i = 1:n
+            @inbounds A[i] ⊻= u << 12*i
+        end
+    end
+    A
+end
+
+# A::Array{UInt128} will match the specialized method above
+function rand!(r::MersenneTwister, A::Base.BitIntegerArray)
+    n = length(A)
+    T = eltype(A)
+    n128 = n * sizeof(T) ÷ 16
+    rand!(r, unsafe_wrap(Array, convert(Ptr{UInt128}, pointer(A)), n128))
+    for i = 16*n128÷sizeof(T)+1:n
+        @inbounds A[i] = rand(r, T)
+    end
+    A
+end
+
+### seeding
+
+#### make_seed()
+
+# make_seed produces values of type Array{UInt32}, suitable for MersenneTwister seeding
+function make_seed()
+    try
+        return rand(RandomDevice(), UInt32, 4)
+    catch
+        println(STDERR,
+                "Entropy pool not available to seed RNG; using ad-hoc entropy sources.")
+        seed = reinterpret(UInt64, time())
+        seed = hash(seed, UInt64(getpid()))
+        try
+            seed = hash(seed, parse(UInt64,
+                                    readstring(pipeline(`ifconfig`, `sha1sum`))[1:40], 16))
+        end
+        return make_seed(seed)
+    end
+end
+
+function make_seed(n::Integer)
+    n < 0 && throw(DomainError(n, "`n` must be non-negative."))
+    seed = UInt32[]
+    while true
+        push!(seed, n & 0xffffffff)
+        n >>= 32
+        if n == 0
+            return seed
+        end
+    end
+end
+
+#### srand()
+
+srand(r::MersenneTwister) = srand(r, make_seed())
+srand(r::MersenneTwister, n::Integer) = srand(r, make_seed(n))
+
+function srand(r::MersenneTwister, seed::Vector{UInt32})
+    copy!(resize!(r.seed, length(seed)), seed)
+    dsfmt_init_by_array(r.state, r.seed)
+    mt_setempty!(r)
+    return r
+end
+
+##### GLOBAL_RNG
+
+function dsfmt_gv_srand()
+    # Temporary fix for #8874 and #9124: update global RNG for Rmath
```

Comment:
cf. https://github.com/JuliaLang/julia/pull/19980, which removes this fix, but whose merge preconditions are not satisfied.